src/components/close-button/CloseButton.tsx

```diff
@@ -0,0 +1,25 @@
+import * as React from 'react';
+import classNames from 'classnames';
+
+import Button, { ButtonType } from '../button';
+import IconClose from '../../icons/general/IconClose';
+import { bdlGray62 } from '../../styles/variables';
+
+import './CloseButton.scss';
+
+export interface CloseButtonProps {
+    /** Custom class for the close button */
+    className?: string;
+    /** onClick handler for the close button */
+    onClick?: Function;
```

Comment:
I love that you are using Typescript! Typescript brother! 😄  Anyway, in my experience, it is good to make types as specific as possible (like `any` is the worst type, `object` is pretty bad too, etc. Those are really escape hatches.) In this case, that means that you could upgrade `Function` to `()=>void` if you like :)